wfcGen.py

```diff
@@ -0,0 +1,287 @@
+from PIL import Image
+import random
+
+res = 0
+WM = False # 0 - no, 1 - yes, WithMove 
+cells = {}
+cropped = []
+for ins in range(4):
+    cropped.append([])
+    if WM == 1:
+        field = Image.open("image.png")
+        
+        SX = field.size[0]
+        SY = field.size[1]
+
+        copy = Image.new("RGB", (SX*2, SY*2))
+        copy.paste(field, (0, 0))
+        copy.paste(field, (SX, 0))
+        copy.paste(field, (0, SY))
+        copy.paste(field, (SX, SY))
+        field = copy
+
+        for y in range(SY):
+            cropped[ins].append([])
+            for x in range(SX):
+                cropped[ins][y].append(field.crop((x, y, x+res, y+res)))
+    else:
+        field = Image.open("image.png")
+        field = field.rotate(90*ins)
+        SX = field.size[0]//res*res
+        SY = field.size[1]//res*res
+        field = field.crop((0, 0, SX, SY))
+
+        for y in range(0, SY, res):
+            cropped[ins].append([])
+            for x in range(0, SY, res):
+                cropped[ins][y//res].append(field.crop((x, y, x+res, y+res)))

```

Comment:
There is a significant amount of duplicated code here. From trivial things, such as `field = Image.open("image.png")`, which can be just moved out, to a little more complex statements, such as the last pair of for loops, which, I belive, you can slightly rewrite to reduce code duplication